### PR TITLE
Install qt5 soundsources to a different folder to avoid crashes, fixes lp1774639

### PR DESCRIPTION
**Pull request by daschuer**
https://bugs.launchpad.net/mixxx/+bug/1774639

This fixes the bug for Linux. I think there is something missing for the Windows and Mac OS installer.

### CODE REVIEW

**Comment by daschuer** [comment]
It is really annoying to have a crash every time I change from QT4 to 5 and back. 
This PR solves this. Can we merge it, at least as an intermediate solution?

**Comment by daschuer** [comment]
Now it fixes also https://bugs.launchpad.net/mixxx/+bug/1711173
The installations for mac OS and Windows are still missing. 
Can one explain how to make it QT5 conditional?

**Comment by Be-ing** [comment]
This feels like a hacky solution. Why isn't the old plugin overwritten by the new one?

**Comment by Be-ing** [comment]
Maybe [this bug](https://bugs.launchpad.net/mixxx/+bug/1406937) is related?

**Comment by daschuer** [comment]
It is not related. The number of soundsources depends on the compile options. So it may happen, that the user has installed a build with more soundsources. These will stick in these folders as intended. 

An other issue is that the user might use Mixxx from the build tree, which used a different qt version than the original installed one. In this case Mixxx tries to load the plug-ins from the install path as well. (My case)

**Comment by daschuer** [comment]
For this PR we need to test and probably fix the windows and Mac installer. Who is able make the installers Qt version aware? Alternative, we can ignore the issue and install master alwaxs to qt5 and 2.1 always to qt4 path.

**Comment by daschuer** [comment]
@sblaisot:What is your opinion here? Is it difficult to make the install location qt version aware on windows?

**Comment by sblaisot** [comment]
The installer already takes whatever subdir exist in build_dir so if the subdir is calles pluginsqt5 instead of plain plugins it will take that and install as-is on target machine: https://github.com/mixxxdj/mixxx/blob/master/src/SConscript#L712,L716

So all you have to do is to build in the right sudbir here:
https://github.com/mixxxdj/mixxx/blob/master/src/SConscript#L567,L570

You already have an example of qt4/5 logic here:
https://github.com/mixxxdj/mixxx/blob/master/src/SConscript#L328,L343

**Comment by rryan** [comment]
I think Windows is mostly unaffected by this bug because the WiX installer will not install Mixxx on top of an old install -- it requires you to run the uninstaller first, which will delete the plugins.

**Comment by daschuer** [comment]
Ok, now it works for Linux only. No crash anymore. 
Linux uses now: 

    /usr/local/lib/mixxx/plugins/soundsourceqt5/
    /usr/lib/mixxx/plugins/soundsourceqt5/
    ~/.mixxx/plugins/soundsourceqt5/

all other platforms are not effected. 

We "only" have to change: 
https://www.mixxx.org/wiki/doku.php/add-ons
the manual is not effected.

The linux packagers are also not effected because at least for ubuntu they do not ship soundsource plugins.

**Comment by daschuer** [comment]
re-added the vamp fix

File: src/sources/soundsourceproxy.cpp [hunk 1]
```diff
@@ -88,7 +95,7 @@ QList<QDir> getSoundSourcePluginDirectories() {
     }
 #elif __WINDOWS__
```
Comment (rryan) [inline src/sources/soundsourceproxy.cpp]: can we leave Windows unchanged?

**Comment by rryan** [comment]
cool, thanks

**Pull request merged**